Cross compiling openblas 0.3.29 version with Android NDK r27c，"kernel/arm/KERNEL.armv8" cannot be read

I tried multiple times to compile using the official documentation **Building for ARMV8**, but failed. Later, I switched to **cmake-gui** to complete the configuration and generate the build files.
### **Compile environment**
Install Android NDK r27c on Ubuntu 22.04, compile to armv8a-Linux-android, API=30

### **Parameters Set**
Set parameters using cmake gui
**Configue Option** using/home/ad/Android/SDK/ndk/27.2.12479018/build/cmake/android.toolchain.cmake
**Other options**
CMAKE_FLAGS   --target=armv8-none-linux-androideabi30  
CMAKE_INSTALL_PREFIX =/home/ad/dev-lib/OpenBLAS/myinstall
BUILD_SHARED_LIB =1
NOFROTRAN=1
BINARY=64
ARCH=arm64
ONLY_CBLAS=1

### **Problem phenomenon**
CMake Error at cmake/utils.cmake:23 (file):
  file STRINGS file "/home/ad/dev-lib/OpenBLAS/kernel/arm/KERNEL.armv8"
  cannot be read.
Call Stack (most recent call first):
  kernel/CMakeLists.txt:13 (ParseMakefileVars)
  kernel/CMakeLists.txt:1374 (build_core)

**### Question**
According to the configuration options, it should be from**/home/ad/dev lib/OpenBLAS/kernel/arm64/KERNEL.armv8** instead of **arm/KERNEL.armv8**. How can I modify the configuration?

[cmake-gui output.txt](https://github.com/user-attachments/files/21215451/cmake-gui.output.txt)

After adding two parameters, ANDROID_ABI=arm64-v8a and ANDROID_PLATFORM=android-30_LEVEL, I tried two methods: cmake gui and scripting.
**Cmake gui** 
Cmake gui still reports the same error,not reading`/kernel/arm64/KERNEL.ARMV8`, still reading`/kernel/arm/KERNEL.ARMV8`,`/kernel/arm/KERNEL.ARMV8`does not exist. Did the compilation document not pass the parameter `BINARY=64` to the relevant compilation file, resulting in incorrect differentiation between arm and arm64.
`CMake Error at cmake/utils.cmake:31 (file):
  file STRINGS file "/home/ad/dev-lib/OpenBLAS1/kernel/arm/KERNEL.ARMV8"
  cannot be read.
Call Stack (most recent call first):
  kernel/CMakeLists.txt:13 (ParseMakefileVars)
  kernel/CMakeLists.txt:1391 (build_core)
`

**Scripting** 
Scripting still  reports the same error:cannot open crtbegin_dynamic.o and crtend_android.o
`/home/ad/Android/Sdk/ndk/27.2.12479018/toolchains/llvm/prebuilt/linux-x86_64/sysroot -fuse-ld=lld -L/home/ad/Android/Sdk/ndk/27.2.12479018/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/aarch64-linux-android/30  -lm     -o .o
ld.lld: error: cannot open crtbegin_dynamic.o: No such file or directory
ld.lld: error: cannot open crtend_android.o: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

--- a/docs/install.md
+++ b/docs/install.md
@@ -623,14 +623,22 @@ Note: using `TARGET=CORTEXA57` in place of `ARMV8` will pick up better
 optimized routines. Implementations for the `CORTEXA57` target are compatible
 with all other `ARMV8` targets.
 
-Note: for NDK 23b, something as simple as:
+Note: for NDK 23b and later, something as simple as:
 ```bash
 export PATH=/opt/android-ndk-r23b/toolchains/llvm/prebuilt/linux-x86_64/bin/:$PATH
-make HOSTCC=gcc CC=/opt/android-ndk-r23b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android31-clang ONLY_CBLAS=1 TARGET=ARMV8
+make HOSTCC=gcc CC=/opt/android-ndk-r23b/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android31-clang ONLY_CBLAS=1 TARGET=ARMV8 RANLIB=echo
 ```
 appears to be sufficient on Linux. On OSX, setting AR to the ar provided in the
 "bin" path of the NDK (probably `llvm-ar`) is also necessary.
 
+If you prefer building with CMake, running 
+```bash
+cmake -DANDROID_ABI=arm64-v8a -DTARGET=ARMV8 -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk-r27/build/cmake/android.toolchain.cmake -DNOFORTRAN=1 -DANDROID_PLATFORM=android-23 ..
+cmake --build .
+```
+in your build directory should work (be sure to adjust the toolchain_file argument according to where you installed the NDK, and the ANDROID_PLATFORM
+according to the minimum version of Android you want to support. (If you leave out the ANDROID_PLATFORM parameter, the build will fail with an error 
+message about a missing declaration or missing header file complex.h)
 
 ??? note "Alternative build script for 3 architectures"
 

